File: CSharp/First Test/ActivationFunction.cs
using System;

namespace Neural_Network
{
    interface ActivationFunction
    {
        double func(double x);
        double funcPrime(double x);
    }


    public class Relu : ActivationFunction
    {
        public double func(double x)
        {
            return Math.Max(x, 0);
        }

        public double funcPrime(double x)
        {
            return x > 0 ? 1 : 0;
        }
    }

    public class Sigmoid : ActivationFunction
    {
        public double func(double x)
        {
            return 1 / (1 + Math.Exp(-x));
        }

        public double funcPrime(double x)
        {
            return func(x) * (1 - func(x));
        }
    }
}

File: Rust/neural-network/src/activation_function/activation_function.rs
use ndarray::Array;

pub struct ActivationFunction {
    pub f: fn(f64) -> f64,
    pub d: fn(f64) -> f64,
}

impl ActivationFunction {
    pub fn function<D>(&self, x: &Array<f64, D>) -> Array<f64, D>
    where
        D: ndarray::Dimension,
    {
        x.mapv(self.f)
    }

    pub fn derivative<D>(&self, x: &Array<f64, D>) -> Array<f64, D>
    where
        D: ndarray::Dimension,
    {
        x.mapv(self.d)
    }
}

pub static SIGMOID: ActivationFunction = ActivationFunction {
    f: (|x: f64| {
        if x < -45.0 {
            0.0
        } else if x > 45.0 {
            1.0
        } else {
            1.0 / (1.0 + (-x).exp())
        }
    }),
    d: (|x: f64| {
        if x < -45.0 {
            0.0
        } else if x > 45.0 {
            0.0
        } else {
            let y = 1.0 / (1.0 + (-x).exp());
            y * (1.0 - y)
        }
    }),
};

pub static RELU: ActivationFunction = ActivationFunction {
    f: (|x: f64| x.max(0.0)),
    d: (|x: f64| if x > 0.0 { 1.0 } else { 0.0 }),
};
